sdcm/cluster.py

```diff
@@ -2144,6 +2145,29 @@ def offline_install_scylla(self, unified_package, nonroot):
             """)
             self.remoter.run('sudo bash -cxe "%s"' % install_cmds)
 
+    def web_install_scylla(self, scylla_version=None) -> None:
+        """
+        Install scylla by Scylla Web Installer Script. Try to use install the latest
+        unstable build for the test branch, generally it should install same version
+        as the unstable scylla_repo.
+
+        https://github.com/scylladb/scylla-web-install
+        """
+        version = 'nightly-master'  # default value
+
+        # Try to get the major version from the branch name, it will only be used when scylla_version isn't assigned.
+        # It can be switched to RELEASE_BRANCH from upstream job
+        git_branch = os.environ.get('GIT_BRANCH')  # origin/branch-4.5
+
+        logging.info(scylla_version)
+        logging.info(git_branch)
+        if scylla_version or git_branch:
+            found = re.match(r'\D+(\d+.\d+)', scylla_version if scylla_version else git_branch)
+            if found:
+                version = f'nightly-{found.groups()[0]}'
```

Comment:
What do you think about removing second `if` for `scylla_version` and `git_branch`?

Something like following:

```python
if match := re.match(..., scylla_version or git_branch):
    scylla_version = f"nightly-{match.group(1)}"
else:
    scylla_version = self.params.get("scylla_version")
```

And I consider to move this logic to `sct_config.py`: anyway you need to tweak logic related `scylla_version` parameter in this file too.

https://github.com/scylladb/scylla-cluster-tests/blob/5099f7d205ff41e0b701dd441d617be35dc862e8/sdcm/sct_config.py#L1392

Reply:
btw, `\D+(\d+.\d+)` regexp is not pretty accurate: it'll not handle strings like `4.4.0`.

What about `\D*(\d+\.\d+)` ?

Reply:
> What do you think about removing second `if` for `scylla_version` and `git_branch`?

Good idea.

> Something like following:
> 
> ```python
> if match := re.match(..., scylla_version or git_branch):
>     scylla_version = f"nightly-{match.group(1)}"
> else:
>     scylla_version = self.params.get("scylla_version")
> ```
> 
> And I consider to move this logic to `sct_config.py`: anyway you need to tweak logic related `scylla_version` parameter in this file too.

The parsed version in web_install_scylla() is different from the scylla_version parameter in jenkins job or in sct config.

I didn't add a new parameter like web_install_version, otherwise we need to assign different value from scylla-pkg in trigger jobs for each releases.

Here I'm trying to parse the release version from scylla_version, and passed the new version parameter (eg: nightly-4.5) to web installer script.

> https://github.com/scylladb/scylla-cluster-tests/blob/5099f7d205ff41e0b701dd441d617be35dc862e8/sdcm/sct_config.py#L1392

Reply:
> btw, `\D+(\d+.\d+)` regexp is not pretty accurate: it'll not handle strings like `4.4.0`.
> 
> What about `\D*(\d+\.\d+)` ?

You are right, fixed.